Cannot compile with disable_3d=yes on macOS for Window x86_64

### Godot version

4.0.dev(bcc061e)

### System information

macOS v13.0

### Issue description

Invoke `scons p=windows target=template_release tools=no use_lto=yes use_mingw=yes arch=x86_64 bits=64 disable_3d=yes` give following error.

```sh
[ 99%] /opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(register_types.windows.template_release.x86_64.o):register_types:(.text+0x4a5): undefined reference to `Node3D::_bind_methods()'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(register_types.windows.template_release.x86_64.o):register_types:(.text+0xa1c): undefined reference to `Node3D::_bind_methods()'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(register_types.windows.template_release.x86_64.o):register_types:(.text+0xa2d): undefined reference to `Node3D::_bind_methods()'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(register_types.windows.template_release.x86_64.o):register_types:(.rdata$.refptr._ZN6Node3D13_bind_methodsEv[.refptr._ZN6Node3D13_bind_methodsEv]+0x0): undefined reference to `Node3D::_bind_methods()'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text+0x17a): undefined reference to `Node3D::Node3D()'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text+0x57e): undefined reference to `Node3D::set_visible(bool)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text+0x5cd): undefined reference to `Skeleton3D::get_bone_parent(int) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text+0x61a): undefined reference to `Skeleton3D::set_bone_pose_rotation(int, Quaternion const&)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text+0x6c6): undefined reference to `Node3D::set_transform(Transform3D const&)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text+0x6d3): undefined reference to `Node3D::set_visible(bool)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text+0x228c): undefined reference to `Skeleton3D::find_bone(String const&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN6Node3D14_notificationvEib[_ZN6Node3D14_notificationvEib]+0x1b): undefined reference to `Node3D::_notification(int)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN6Node3D14_notificationvEib[_ZN6Node3D14_notificationvEib]+0x25): undefined reference to `Node3D::_notification(int)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN6Node3D14_notificationvEib[_ZN6Node3D14_notificationvEib]+0x64): undefined reference to `Node3D::_notification(int)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN6Node3D14_notificationvEib[_ZN6Node3D14_notificationvEib]+0x7a): undefined reference to `Node3D::_notification(int)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK6Node3D21_property_get_revertvERK10StringNameR7Variant[_ZNK6Node3D21_property_get_revertvERK10StringNameR7Variant]+0x3): undefined reference to `Node3D::_property_get_revert(StringName const&, Variant&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK6Node3D21_property_get_revertvERK10StringNameR7Variant[_ZNK6Node3D21_property_get_revertvERK10StringNameR7Variant]+0x14): undefined reference to `Node3D::_property_get_revert(StringName const&, Variant&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK10OpenXRHand21_property_get_revertvERK10StringNameR7Variant[_ZNK10OpenXRHand21_property_get_revertvERK10StringNameR7Variant]+0x3): undefined reference to `Node3D::_property_get_revert(StringName const&, Variant&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK10OpenXRHand21_property_get_revertvERK10StringNameR7Variant[_ZNK10OpenXRHand21_property_get_revertvERK10StringNameR7Variant]+0x14): undefined reference to `Node3D::_property_get_revert(StringName const&, Variant&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK6Node3D21_property_can_revertvERK10StringName[_ZNK6Node3D21_property_can_revertvERK10StringName]+0x3): undefined reference to `Node3D::_property_can_revert(StringName const&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK6Node3D21_property_can_revertvERK10StringName[_ZNK6Node3D21_property_can_revertvERK10StringName]+0x14): undefined reference to `Node3D::_property_can_revert(StringName const&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK10OpenXRHand21_property_can_revertvERK10StringName[_ZNK10OpenXRHand21_property_can_revertvERK10StringName]+0x3): undefined reference to `Node3D::_property_can_revert(StringName const&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK10OpenXRHand21_property_can_revertvERK10StringName[_ZNK10OpenXRHand21_property_can_revertvERK10StringName]+0x14): undefined reference to `Node3D::_property_can_revert(StringName const&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK6Node3D19_validate_propertyvER12PropertyInfo[_ZNK6Node3D19_validate_propertyvER12PropertyInfo]+0x3): undefined reference to `Node3D::_validate_property(PropertyInfo&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK6Node3D19_validate_propertyvER12PropertyInfo[_ZNK6Node3D19_validate_propertyvER12PropertyInfo]+0x14): undefined reference to `Node3D::_validate_property(PropertyInfo&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK10OpenXRHand19_validate_propertyvER12PropertyInfo[_ZNK10OpenXRHand19_validate_propertyvER12PropertyInfo]+0x3): undefined reference to `Node3D::_validate_property(PropertyInfo&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZNK10OpenXRHand19_validate_propertyvER12PropertyInfo[_ZNK10OpenXRHand19_validate_propertyvER12PropertyInfo]+0x14): undefined reference to `Node3D::_validate_property(PropertyInfo&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN6Node3D18_initialize_classvEv[_ZN6Node3D18_initialize_classvEv]+0xfb): undefined reference to `Node3D::_bind_methods()'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN6Node3D18_initialize_classvEv[_ZN6Node3D18_initialize_classvEv]+0x10c): undefined reference to `Node3D::_bind_methods()'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN10OpenXRHand18_initialize_classvEv[_ZN10OpenXRHand18_initialize_classvEv]+0x1ee): undefined reference to `Node3D::_bind_methods()'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN10OpenXRHand18_initialize_classvEv[_ZN10OpenXRHand18_initialize_classvEv]+0x1ff): undefined reference to `Node3D::_bind_methods()'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN10OpenXRHand14_notificationvEib[_ZN10OpenXRHand14_notificationvEib]+0x3a): undefined reference to `Node3D::_notification(int)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN10OpenXRHand14_notificationvEib[_ZN10OpenXRHand14_notificationvEib]+0x49): undefined reference to `Node3D::_notification(int)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN10OpenXRHand14_notificationvEib[_ZN10OpenXRHand14_notificationvEib]+0x10b): undefined reference to `Node3D::_notification(int)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.text$_ZN10OpenXRHand14_notificationvEib[_ZN10OpenXRHand14_notificationvEib]+0x11a): undefined reference to `Node3D::_notification(int)'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.rdata$.refptr._ZNK6Node3D18_validate_propertyER12PropertyInfo[.refptr._ZNK6Node3D18_validate_propertyER12PropertyInfo]+0x0): undefined reference to `Node3D::_validate_property(PropertyInfo&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.rdata$.refptr._ZNK6Node3D20_property_can_revertERK10StringName[.refptr._ZNK6Node3D20_property_can_revertERK10StringName]+0x0): undefined reference to `Node3D::_property_can_revert(StringName const&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.rdata$.refptr._ZNK6Node3D20_property_get_revertERK10StringNameR7Variant[.refptr._ZNK6Node3D20_property_get_revertERK10StringNameR7Variant]+0x0): undefined reference to `Node3D::_property_get_revert(StringName const&, Variant&) const'
/opt/homebrew/Cellar/mingw-w64/10.0.0_3/toolchain-x86_64/bin/x86_64-w64-mingw32-ld: modules/libmodule_openxr.windows.template_release.x86_64.a(openxr_hand.windows.template_release.x86_64.o):openxr_hand.cp:(.rdata$.refptr._ZN6Node3D13_notificationEi[.refptr._ZN6Node3D13_notificationEi]+0x0): undefined reference to `Node3D::_notification(int)'
collect2: error: ld returned 1 exit status
scons: *** [bin/godot.windows.template_release.x86_64.exe] Error 1
scons: building terminated because of errors.
[Time elapsed: 00:04:12.245]
```

### Steps to reproduce

N/A

### Minimal reproduction project

N/A

--- a/modules/openxr/config.py
+++ b/modules/openxr/config.py
@@ -1,6 +1,6 @@
 def can_build(env, platform):
     if platform in ("linuxbsd", "windows", "android"):
-        return env["openxr"]
+        return env["openxr"] and not env["disable_3d"]
     else:
         # not supported on these platforms
         return False

